kinematics_interface/include/kinematics_interface/kinematics_interface_base.hpp

```diff
@@ -39,46 +41,49 @@ class KinematicsBaseClass
   /**
      * \brief Convert Cartesian delta-x to joint delta-theta, using the Jacobian.
      * \param[in] joint_pos joint positions of the robot in radians
-     * \param[in] delta_x_vec input Cartesian deltas (x, y, z, wx, wy, wz)
-     * \param[out] delta_theta_vec output vector with joint states
+     * \param[in] delta_x input Cartesian deltas (x, y, z, wx, wy, wz)
+     * \param[in] link_name the link name at which delta_x is applied
+     * \param[out] delta_theta outputs joint deltas
      * \return true if successful
      */
   virtual bool convert_cartesian_deltas_to_joint_deltas(
-    const std::vector<double> & joint_pos, const std::vector<double> & delta_x_vec,
-    const std::string & link_name, std::vector<double> & delta_theta_vec) = 0;
+    const Eigen::VectorXd & joint_pos, const Eigen::Matrix<double, 6, 1> & delta_x,
```

Comment:
```suggestion
    const Eigen::VectorXd & joint_pos, const Eigen::Vector6d & delta_x,
```

Can probably make the same change in several other places

Reply:
I don't think `Eigen::Vector6d ` is a standard type in Eigen.